webapp/app/routes/app/l2-cancellations/$id/_route.tsx

```diff
@@ -226,17 +252,28 @@ export default function L2Cancellation() {
             <CardTitle>Execute Actions</CardTitle>
           </CardHeader>
           <CardContent className="flex flex-col space-y-3">
-            <ExecL2VetoForm
-              proposalId={proposal.id}
-              guardiansAddress={guardiansAddress}
-              signatures={signatures}
-              threshold={necessarySignatures}
-              proposal={proposal}
-              disabled={execDisabled}
-              calls={calls}
-            >
-              Execute Veto
-            </ExecL2VetoForm>
+            {proposal.status === "ACTIVE" && (
+              <>
+                <ExecL2VetoForm
+                  proposalId={proposal.id}
+                  guardiansAddress={guardiansAddress}
+                  signatures={signatures}
+                  threshold={necessarySignatures}
+                  proposal={proposal}
+                  disabled={execDisabled}
+                  calls={calls}
+                >
+                  Execute Veto
+                </ExecL2VetoForm>
+
+                {!isExactNonce && (
+                  <p className="text-red-500 text-s">
+                    Nonce does not match with contract. Maybe some other veto has to be executed
+                    before.
+                  </p>
+                )}
```

Comment:
Love this. Perhaps it's better if we show the user not as an error but instead a warning with an orange text?